src/main/java/com/squareup/javapoet/CodeWriter.java

```diff
@@ -456,7 +456,7 @@ private ClassName stackClassName(int stackDepth, String simpleName) {
    */
   CodeWriter emitAndIndent(String s) throws IOException {
     boolean first = true;
-    for (String line : s.split("\n", -1)) {
+    for (String line : s.split("\r?\n", -1)) {
```

Comment:
Javapoet is a Java 8 project, so we might be able to use the linebreak matcher `\\R` instead.

Reply:
I'll update the PR. Thanks